guardian/elastic-search-monitor
# chore(deps): Update dependencies and move to NPM

## What does this change?
We're currently using Yarn v1, which is EOL; move to NPM.

This has been done by recreating a GuCDK project via:

```sh
npx @guardian/cdk@latest new --package-manager npm --stage PROD --app elastic-search-monitor --stack deploy
```

Consequently this updates all dependencies, including:
  - AWS CDK (transitive) to 2.145.0 which includes CloudWatch Alarm tagging.
  - Jest (direct) to 29.7.0, a major update that brings snapshot format changes. To reduce the snapshot diff, configure Jest to use the old snapshot format. See https://jestjs.io/blog/2022/08/25/jest-29.
  - Transitive dependencies are updated, resolving the open [alerts](https://github.com/guardian/elastic-search-monitor/security/dependabot), at least [`npm audit`](https://docs.npmjs.com/cli/v9/commands/npm-audit) yields no issues.

## How to test
These are infrastructure changes - checking the snapshot diff should suffice. It should have:
  - One significant change; tags on alarms.
  - One minor change; Jest snapshot formatting.

For ease, the former can be viewed in the first commit, and the latter in the second commit.
